svgnative/ports/cairo/CairoSVGRenderer.h

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 suzuki toshiya <mpsuzuki@hiroshima-u.ac.jp>. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+#ifndef SVGViewer_CairoSVGRenderer_h
+#define SVGViewer_CairoSVGRenderer_h
+
+#include <list>
+#include "SVGRenderer.h"
+#include "cairo.h"
+
+namespace SVGNative
+{
+
+  // SkiaSVGPath object is able to be amended, but Cairo has no API to append something
+  // to existing cairo_path_t object. Thus, we hold cairo_t object. Its surface could
+  // be retrieved in later, by cairo_get_target().
+
+class CairoSVGPath final : public Path
+{
+public:
+    CairoSVGPath();
+    ~CairoSVGPath();
+
+    void Rect(float x, float y, float width, float height) override;
+    void RoundedRect(float x, float y, float width, float height, float cornerRadius) override;
+    void Ellipse(float cx, float cy, float rx, float ry) override;
+
+    void MoveTo(float x, float y) override;
+    void LineTo(float x, float y) override;
+    void CurveTo(float x1, float y1, float x2, float y2, float x3, float y3) override;
+    void CurveToV(float x2, float y2, float x3, float y3) override;
+    void ClosePath() override;
+
+    cairo_t* mPath{}; // FIXME: rename to mPathContext in future, because it is not a path object.
```

Comment:
Fix the FIXME in this patch?

Reply:
OK, I would do... Trying mPathContext, I felt it's slightly too long. "mPathCtx" is acceptable?

Reply:
sure